components/pagination/src/pagination.js

```diff
@@ -38,6 +38,11 @@ const Pagination = ({
         pageSize,
         total,
     })
+    const showPageSelect =
+        !hidePageSelect &&
+        typeof pageCount === 'number' &&
+        pageCount > 1 &&
+        pageCount <= 2000
```

Comment:
[Magic number](https://rules.sonarsource.com/typescript/RSPEC-109), I'd say we either put that in a const or a prop (with default value)